MapboxCoreNavigation/RouteController.swift

```diff
@@ -282,6 +282,17 @@ open class RouteController: NSObject, Router {
         let isCloseToCurrentStep = location.isWithin(radius, of: routeProgress.currentLegProgress.currentStep)
         return isCloseToCurrentStep
     }
+    
+    /**
+     Advances the leg index.
+     
+     This is a convienence method provided to advance the leg index of any given router without having to worry about the internal data structure of the router.
+     */
+    public func advanceLegIndex(location: CLLocation) {
+        precondition(!routeProgress.isFinalLeg, "Can not increment leg index beyond final leg.")
+        routeProgress.legIndex += 1
```

Comment:
Technically, it is possible for application code to reach in and set `RouteProgress.legIndex` to anything at any time. Prior to MapboxNavigationNative integration, doing so would presumably skip legs seamlessly. But now doing so could cause RouteProgress and the navigator to get out of sync. Perhaps PortableRouteController should observe its RouteProgress object’s `legIndex` key path and update the navigator in response.

Applications intentionally mucking with the leg index may be a bit of an edge case, so we can treat it as tail work. But I think accounting for this case will make the code much more robust, so we should definitely consider it.

Reply:
It may sound absurd that an application might set the leg index to an arbitrary value, but that could happen in an application that simulates routes and hooks up buttons to navigate back and forth among steps or legs.

Reply:
Going to treat this as tail work.